src/interceptors/on-request-refresh-token.ts

```diff
@@ -0,0 +1,52 @@
+import { AxiosError, InternalAxiosRequestConfig } from 'axios';
+import { RefreshTokenOptions } from '../types';
+import { checkIsTokenExpired } from '../utils';
+
+let refreshTokenRequest: Promise<string> | null;
+
+export const onRequestRefreshToken =
```

Comment:
After another look I think we should include the `Interceptor` suffix into the name for consistency with other interceptors:

```suggestion
export const onRequestRefreshTokenInterceptor =
```